lib/service_order_listener.rb

```diff
@@ -0,0 +1,44 @@
+class ServiceOrderListener
+  attr_accessor :messaging_client_options, :client
+
+  def initialize(messaging_client_options = {})
+    self.messaging_client_options = default_messaing_options.merge(messaging_client_options)
+  end
+
+  def run
+    Thread.new do
+      self.client = ManageIQ::Messaging::Client.open(messaging_client_options)
+
+      client.subscribe_messages({
+        :service   => "platform.topological-inventory.task-output-stream",
+        :max_bytes => 500_000
+      }) do |messages|
+        messages.each do |msg|
+          process_message(msg)
+        end
+      end
+    end
+  ensure
+    client&.close
+    self.client = nil
+  end
+
+  private
+
+  def process_message(msg)
+    if msg.payload["state"] == "completed"
+      item = OrderItem.where(:topology_task_ref => msg.payload["task_id"]).first
+      item.state = 'Order Completed?'
```

Comment:
@eclarizio We have a class called ProgressMessage connected to an order item, we should take all messages and store it in the progress message class.
@eclarizio We also need to handle failure/error cases here.

Reply:
@eclarizio Also need to have an exception if the OrderItem is not found, if they get deleted from the database.